src/multisig/multisigTransactionsOutcomeParser.ts

```diff
@@ -0,0 +1,278 @@
+
+        // First, look in "logs":
+        eligibleEvents.push(
+            ...transactionOnNetwork.logs.events.filter((event) => event.identifier === eventIdentifier),
+        );
+
+        // Then, look in "logs" of "contractResults":
+        for (const result of transactionOnNetwork.smartContractResults) {
+            eligibleEvents.push(...result.logs.events.filter((event) => event.identifier === eventIdentifier));
+        }
+
+        if (eligibleEvents.length === 0) {
+            return null;
+        }
+
+        const [event] = eligibleEvents;
+        const data = Buffer.from(event.data).toString();
+        const parts = data.split("@").filter(Boolean);
+
+        let returnCode = "ok";
+        if (parts.length > 0) {
+            returnCode = Buffer.from(parts[0], "hex").toString() || "ok";
+        }
+
+        const returnDataParts = parts.slice(1).map((part) => Buffer.from(part, "hex"));
+
+        return new SmartContractCallOutcome({
+            function: transactionOnNetwork.function,
+            returnCode: returnCode,
+            returnMessage: returnCode,
+            returnDataParts: returnDataParts,
+        });
+    }
+
+    /**
+     * Parses the outcome of a query to get the multisig contract's pending actions
+     * @param queryResponse The query response
+     * @returns The list of pending action IDs
+     */
+    parsePendingActionIds(queryResponse: string[]): number[] {
+        try {
+            if (!queryResponse || queryResponse.length === 0) {
+                return [];
+            }
+
+            // Assuming each element in the response is a base64 encoded action ID
+            return queryResponse.map((item) => {
+                const buffer = Buffer.from(item, "base64");
+                return parseInt(buffer.toString("hex"), 16);
+            });
+        } catch (error) {
+            throw new Error(`Error parsing pending action IDs: ${error}`);
+        }
+    }
+
+    /**
+     * Parses the outcome of a query to get the multisig contract's board members
+     * @param queryResponse The query response
+     * @returns The list of board member addresses
+     */
+    parseBoardMembers(queryResponse: string[]): Address[] {
+        if (!queryResponse || queryResponse.length === 0) {
+            return [];
+        }
+
+        return queryResponse.map((item) => {
+            const buffer = Buffer.from(item, "base64");
+            return Address.newFromHex(buffer.toString("hex"));
+        });
+    }
+
+    /**
+     * Parses the outcome of a query to get the multisig contract's quorum
+     * @param queryResponse The query response
+     * @returns The quorum value
+     */
+    parseQuorum(queryResponse: string[]): number {
+        if (!queryResponse || queryResponse.length === 0) {
+            throw new Err("No return data available");
+        }
+
+        const buffer = Buffer.from(queryResponse[0], "base64");
+        return parseInt(buffer.toString("hex"), 16);
```

Comment:
Same as an earlier comment - or, perhaps, for starters, extract this conversion to a private function here, in the outcome parser.